File: backend/src/main/java/com/grupo1/backend/entities/Comentario.java
package com.grupo1.backend.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "comentarios")
public class Comentario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "nota")
    private int nota;

    @Column(name = "pros")
    private String pros;

    @Column(name = "contras")
    private String contras;

    @Column(name = "opinion")
    private String opinion;

    @ManyToOne
    @JoinColumn(name = "id_user")
    @JsonIgnore
    private User user;

    @ManyToOne
    @JoinColumn(name = "id_producto")
    @JsonBackReference
    private Producto producto;
}

File: frontEnd/src/app/models/comentario.ts
export class Comentario {
  id?: number;
  nota: number;
  pros: string;
  contras: string;
  opinion: string;
  usuario: { id: number };       
  producto: { id: number };      

  constructor(
    nota: number,
    pros: string,
    contras: string,
    opinion: string,
    id_user: number,
    id_producto: number
  ) {
    this.nota = nota;
    this.pros = pros;
    this.contras = contras;
    this.opinion = opinion;
    this.usuario = { id: id_user };
    this.producto = { id: id_producto };
  }
}
